code/sorting/sort_test/sort_test.cpp

```diff
@@ -2,109 +2,163 @@
+using ContainerType = std::deque<_T>;
+#endif
+
+template<typename _Iter1,
+         typename _Iter2,
+         typename _Tp = typename std::iterator_traits<_Iter1>::value_type>
+bool isSame(_Iter1 aBegin, _Iter1 aEnd, _Iter2 bBegin, _Iter2 bEnd) {
+    auto aIt = aBegin;
+    auto bIt = bBegin;
+    for (; aIt != aEnd; ++aIt, ++bIt)
         if (*aIt != *bIt)
             return false;
-    }
+
     return true;
 }
 
-using namespace std;
 TEST_CASE("sort algorithm") {
     void (*psf)(int*, int*);
-    void (*vsf)(vector<int>::iterator, vector<int>::iterator);
+    void (*vsf)(ContainerType<int>::iterator, ContainerType<int>::iterator);
     
-    // substitute search algorithm
+    // substitute sort algorithm
     vsf = merge_sort::mergeSort;
     psf = merge_sort::mergeSort;
-    
+//    vsf = bubbleSort;
+//    psf = bubbleSort;
+//    vsf = insertionSort;
+//    psf = insertionSort;
+//    vsf = selectionSort;
+//    psf = selectionSort;
+//    vsf = quickSort;
+//    psf = quickSort;
+
     SECTION("test empty") {
         const int sz = 0;
         int *arr = new int[sz];
         int *arr_end = arr+sz;
-        vector<int> vec(sz);
+        ContainerType<int> stdContainer(sz);
         
         psf(arr, arr_end);
-        vsf(vec.begin(), vec.end());
+        vsf(stdContainer.begin(), stdContainer.end());
     }
     
     SECTION("test has 1 elem") {
         const int sz = 1;
-        int *arr = new int[sz];
+        int *arr = new int[sz]{1};
         int *arr_end = arr+sz;
-        vector<int> vec(sz);
-        arr[0] = 1;
-        vec[0] = 1;
+        ContainerType<int> stdContainer{1};
         
         psf(arr, arr_end);
-        vsf(vec.begin(), vec.end());
+        vsf(stdContainer.begin(), stdContainer.end());
         
         CHECK(arr[0] == 1);
-        CHECK(vec[0] == 1);
+        CHECK(*stdContainer.begin() == 1);
     }
     
     SECTION("test has 2 elems") {
         const int sz = 2;
         int *arr = new int[sz]{3, 1};
         int *arr_end = arr+sz;
-        vector<int> vec(sz);
-        vec[0] = 3;
-        vec[1] = 1;
+        ContainerType<int> stdContainer{3, 1};
         
         psf(arr, arr_end);
-        vsf(vec.begin(), vec.end());
+        vsf(stdContainer.begin(), stdContainer.end());
         
         CHECK(arr[0] == 1);
         CHECK(arr[1] == 3);
-        CHECK(vec[0] == 1);
-        CHECK(vec[1] == 3);
+        CHECK(*stdContainer.begin() == 1);
+        CHECK(*++stdContainer.begin() == 3);
+    }
+
+    SECTION("test has 3 elems") {
+        const int sz = 3;
+        int *arr = new int[sz]{2, 3, 1};
+        int *arr_end = arr+sz;
+        ContainerType<int> stdContainer{2, 3, 1};
+
+        psf(arr, arr_end);
+        vsf(stdContainer.begin(), stdContainer.end());
+
+        CHECK(arr[0] == 1);
+        CHECK(arr[1] == 2);
+        CHECK(arr[2] == 3);
+        CHECK(*stdContainer.begin() == 1);
+        CHECK(*++stdContainer.begin() == 2);
+        CHECK(*++++stdContainer.begin() == 3);
```

Comment:
Any way you could improve the readability of this (With a comment)?

Reply:
😅 I updated, please review.